### DISCUSSION TITLE
Ability to show Icons in a custom cartersian marker

### DISCUSSION THREAD

**Discussion by Praveen-Android** [patrykandpatrick/vico#881] [Questions] [answered]
### Question

is it possible to customize the marker label to show an icon drawable in addition to text?

### Vico version(s)

2.0.0-alpha.28

**Comment by Gowsky** [comment]
Hi @Praveen-Android. Yes, it's possible. Could you clarify if "custom cartesian marker" refers to a customized instance of `DefaultCartesianMarker`, or a fully custom implementation of the `CartesianMarker` interface?

**Reply by Praveen-Android** [reply]
Thanks for the reply @Gowsky . Here is something that I am looking to achieve:
![Property 1=Manual Entry](https://github.com/user-attachments/assets/e286a9ed-589d-42d1-9050-95db292ed1f4)

Here is what I was able to do so far:
- Customize the indicator using the indicator property of `DefaultCartesianMarker`. 
- Prepare custom label background using `background` property of label `TextComponent` that gets fed in to `DefaultCartesianMarker`. 

Couple things I am not sure on how to do:
- Multiple labels with different styles appended together
- Add an icon (as shown in the image)

From what i understand, its possible to manipulate text using `CartesianMarkerValueFormatter` but not sure about the icon. Would highly appreciate any suggestions/directions in this regards.

Thanks!

**Reply by Praveen-Android** [reply]
@Gowsky Just curious to see if you got a chance to take a look at my reply above. Any direction on this would be super helpful. Appreciate the time!

**Reply by Gowsky** [reply, marked as answer]
@Praveen-Android `CartesianMarkerValueFormatter` expects you to return `CharSequence`. Therefore you can use [`SpannableString`](https://developer.android.com/reference/android/text/SpannableString) with [`ImageSpan`](https://developer.android.com/reference/android/text/style/ImageSpan) to add an icon to your text. Also, please note that [`CartesianMarker`](https://patrykandpatrick.com/vico/api/2.0.0-beta.1/core/com.patrykandpatrick.vico.core.cartesian.marker/-cartesian-marker/?query=interface%20CartesianMarker%20:%20CartesianLayerInsetter%3CCartesianChartModel%3E) is an interface that allows you to draw anything on `Canvas`.

**Reply by Praveen-Android** [reply]
ah ok understood. Will give it a try. Thanks for the reply!

**Reply by Gowsky** [reply]
You're welcome!

**Reply by Praveen-Android** [reply]
@Gowsky fyi, I tried your suggestion and was able to get the desired marker. Thanks much! 

https://github.com/user-attachments/assets/987a8c0f-3da3-4253-a6f1-cd9b0f9836b5

One thing I noticed is that there is quite a bit of margin that is added for markers at the corners. (Image and in the video above). I would like the marker be as close to the point as possible but not go off the edge. Looks like the rectangle is rendered by `CorneredShape` and the triangular tick itself is rendered by `MarkerCorneredShape`. 

So may be i need a custom implementation of `CorneredShape` and `MarkerCorneredShape` could use that? but not sure if thats the way to go. 

**Could you give me any pointers on what pieces I would need to update in probably the outline method of `CorneredShape` to achieve what I am looking for.** 

![image](https://github.com/user-attachments/assets/0acf42cf-ab9e-44ec-b07c-fec6d844fd59)

Thanks!

**Reply by Gowsky** [reply]
Please share your code. This doesn't happen in the sample app. A custom `Shape` shouldn't be necessary here.

**Reply by Praveen-Android** [reply]
Thanks for your inputs. I investigated further and there was a minor issue in my code which was causing this to happen. Its working as expected now. Thanks for your feedback!

**Reply by cosminlinca** [reply]
Hello, @Praveen-Android, how did you managed to implement "Multiple labels with different styles appended together"?

**Reply by Gowsky** [reply]
@cosminlinca, as per my comment, you can do this via `SpannableString`. For example, to adjust the size of a specific part of the label, you'd use `AbsoluteSizeSpan` or `RelativeSizeSpan`.

**Reply by cosminlinca** [reply]
Thank you @Gowsky !

**Reply by Gowsky** [reply]
You're welcome, @cosminlinca!

**Reply by AnuragPandey01** [reply]
Hi! @Gowsky, how to implement for compose multiplatform as SpannableString is android specific?

**Reply by Gowsky** [reply]
Hello, @AnuragPandey01. You'd use `AnnotatedString` for this, but this requires `@Composable` markers, which aren't supported at the time. Thus, you have to draw icons manually by implementing a custom `CartesianMarker`.
